zkvm/src/encoding.rs

```diff
@@ -5,20 +5,19 @@ use byteorder::{ByteOrder, LittleEndian};
 use core::ops::Range;
 use curve25519_dalek::ristretto::CompressedRistretto;
 use curve25519_dalek::scalar::Scalar;
-use std::ops::Deref;
 
 use crate::errors::VMError;
 
 #[derive(Copy, Clone, Debug)]
```

Comment:
1. Let's remove Copy entirely to make all clones explicit (this will break the `::parse` and hint that we probably should pass &mut to closure instead of moving the reader in it).
2. Let's _try_ to remove Clone and see if we can live without it. It would be safer if we never have to clone the reader and can use its specialized APIs to do all the things we need.